packages/cinterop/src/jvm/jni/utils.cpp

```diff
@@ -82,13 +82,12 @@ static std::string string_to_hex(const std::string& message, realm::StringData&
     ret << "error_code = " << error_code << "; ";
     ret << "retcode = " << retcode << "; ";
     ret << "StringData.size = " << str.size() << "; ";
-    ret << "StringData.data = " << str << "; ";
-    ret << "StringData as hex = ";
+    ret << "StringData as hex =";
     for (std::string::size_type i = 0; i < str.size(); ++i)
         ret << " 0x" << std::hex << std::setfill('0') << std::setw(2) << (int)s[i];
     ret << "; ";
-    ret << "in_begin = " << in_begin << "; ";
-    ret << "in_end = " << in_end << "; ";
+    ret << "in_begin = " << (void*) in_begin << "; ";
```

Comment:
why casting to opaque pointer?

Reply:
If treating `in_begin` and `in_end` as `char *` they will be considered as 0-terminated strings and printed to the stream ... in which case we end up including the invalid UTF-8 bytes in the message - Just as for `ret << "StringData.data = " << str << "; ";`. Wan't to avoid having invalid unicode in the exception message as spell out in https://github.com/realm/realm-kotlin/pull/1792#discussion_r1665489727